momento-sdk/src/main/java/momento/sdk/AuthClient.java

```diff
@@ -0,0 +1,44 @@
+package momento.sdk;
+
+import java.util.concurrent.CompletableFuture;
+import momento.sdk.auth.accessControl.DisposableTokenScope;
+import momento.sdk.auth.accessControl.ExpiresIn;
+import momento.sdk.exceptions.InvalidArgumentException;
+import momento.sdk.responses.GenerateDisposableTokenResponse;
+
+public class AuthClient implements IAuthClient {
+  private final TokenClient tokenClient;
+
+  public AuthClient(TokenClient tokenClient) {
```

Comment:
hrm I think the token client is supposed to be a private/internal implementation detail, so we would want a constructor that allows you to construct an AuthClient without already having a TokenClient?